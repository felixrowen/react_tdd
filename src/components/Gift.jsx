import React, { Component } from 'react'
import {
  Form,
  FormGroup,
  FormControl,
  FormLabel,
  Button
} from 'react-bootstrap'

class Gift extends Component{
  constructor() {
    super()
    this.state = {
      person: '',
      present: ''
    }
  }

  render() {
    return(
      <div>
        <Form>
          <FormGroup>
            <FormLabel>Person</FormLabel>
            <FormControl 
              className="input-person"
              onChange={(event) => this.setState({person: event.target.value})}
            />
          </FormGroup>
          <FormGroup>
            <FormLabel>Present</FormLabel>
            <FormControl 
              className="input-present"
              onChange={(event) => this.setState({present: event.target.value})}
            />
          </FormGroup>
        </Form>
        <Button
          className="btn-remove"
          onClick={() => this.props.removeGift(this.props.gift.id)}
        >
          Remove Gift
        </Button>
      </div>
    )
  }
}

export default Gift