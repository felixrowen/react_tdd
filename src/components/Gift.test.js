import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import Gift from './Gift'

configure({adapter: new Adapter()})

describe('Gift', () => {
  const mockRemove = jest.fn()
  const id = 1
  const props = { gift: { id }, removeGift: mockRemove }
  const gift = shallow(<Gift {...props} />)

  it('renders properly', () => {
    expect(toJson(gift)).toMatchSnapshot()
  })

  it('it initializes a person and present in `state`', () => {
    expect(gift.state()).toEqual({ person: '', present: '' })
  })

  describe('when typing into the person input', () => {
    const person = 'Uncle'

    beforeEach(() => {
      gift.find('.input-person').simulate('change', { target: { value: person } }) 
    })

    it('updates the person in `state`', () => {
      expect(gift.state().person).toEqual('Uncle')
    })
  })

  describe('when typing into the present input', () => {
    const present = 'Golf Clubs'

    beforeEach(() => {
      gift.find('.input-present').simulate('change', { target: { value: present } })
    })

    it('updates the present in `state`', () => {
      expect(gift.state().present).toEqual(present)
    })
  })

  describe('when clicking the `remove gift` button', () => {
    beforeEach(() => {
      gift.find('.btn-remove').simulate('click')
    })

    it('calls the removeGift callback', () => {
      expect(mockRemove).toHaveBeenCalledWith(id)
    })
  })

})