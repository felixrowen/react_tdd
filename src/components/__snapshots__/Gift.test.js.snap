// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders properly 1`] = `
<div>
  <Form
    as="form"
    inline={false}
  >
    <FormGroup
      as="div"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Person
      </FormLabel>
      <FormControl
        as="input"
        className="input-person"
        onChange={[Function]}
      />
    </FormGroup>
    <FormGroup
      as="div"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Present
      </FormLabel>
      <FormControl
        as="input"
        className="input-present"
        onChange={[Function]}
      />
    </FormGroup>
  </Form>
  <Button
    active={false}
    className="btn-remove"
    disabled={false}
    onClick={[Function]}
    type="button"
    variant="primary"
  >
    Remove Gift
  </Button>
</div>
`;
